5_analyze_data/notebooks/KS-test_analysis/visualize_ks_test.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Visualize the results of the KS-test
+
+# In[1]:
+
+
+suppressPackageStartupMessages(library(ggplot2))
+suppressPackageStartupMessages(library(dplyr))
+
+
+# In[2]:
+
+
+output_fig_dir <- "figures"
+
+ks_test_fig <- file.path(output_fig_dir, "ks_test_cp_genotype.png")
+actin_distrib_fig <- file.path(output_fig_dir, "top_actin_feature_distrib.png")
+
+
+# In[3]:
+
+
+# Load and process KS-test data
+ks_file <- file.path("data", "nf1_kstest_two_sample_results.csv")
+ks_df <- readr::read_csv(ks_file, col_types = readr::cols(.default="d", Features="c"))
+
+# Create a new column for a directional ks-test statistic
+# Arrange by statistic
+# Split out components of feature name for visualization
+ks_df <- ks_df %>%
+    dplyr::mutate(signed_statistic = statistic * statistic_sign) %>%
+    dplyr::arrange(desc(statistic)) %>%
+    tidyr::separate(
+        Features,
+        into = c(
+            "compartment",
+            "feature_group",
+            "measurement",
+            "channel", 
+            "parameter1", 
+            "parameter2"
+        ),
+        sep = "_",
+        remove = FALSE
+    ) %>%
+    dplyr::mutate(channel_cleaned = channel)
+
+# Clean channel for visualization
+ks_df$channel_cleaned <-
+    dplyr::recode(
+        ks_df$channel_cleaned,
+        "DAPI" = "nuclei",
+        "RFP" = "actin",
+        "GFP" = "ER",
+        .default = "other",
+        .missing="other"
+    )
+
+print(dim(ks_df))
+head(ks_df, 10)
+
+
+# In[4]:
+
+
+# Load feature data
+data_dir <-file.path("..", "..", "..", "4_processing_features", "data")
+cp_file <- file.path(data_dir, "nf1_sc_norm_cellprofiler.csv.gz")
+
+cp_df <- readr::read_csv(
+    cp_file,
+    col_types = readr::cols(
+        .default="d",
+        Metadata_WellRow="c",
+        Metadata_WellCol="c",
+        Metadata_Well="c",
+        Metadata_gene_name="c",
+        Metadata_genotype="c"
+    )
+)
+
+print(dim(cp_df))
+head(cp_df, 3)
+
+
+# ## Visualize KS test
+
+# In[5]:
+
+
+# Determine a bonferroni adjusted alpha value threshold
```

Comment:
Why use this to determine the threshold? Is there a reason or is this the standard?

Reply:
that is correct! we need to adjust our alpha value for something called "multiple test correction". Bonferroni is the simplest and most straightfoward answer. Just divide 0.05 by the number of comparisons!

Reply:
That is very simple! Good to know there are multiple ways of doing this also but this simple version seems to be the best option for this project.